lib/internal/modules/esm/hooks.js

```diff
@@ -65,6 +66,30 @@ let debug = require('internal/util/debuglog').debuglog('esm', (fn) => {
 });
 let importMetaInitializer;
 
+let importAssertionAlreadyWarned = false;
+
+function emitImportAssertionWarning() {
+  if (!importAssertionAlreadyWarned) {
+    importAssertionAlreadyWarned = true;
+    process.emitWarning('Use `importAttributes` instead of `importAssertions`', 'ExperimentalWarning');
+  }
+}
```

Comment:
There's some flavour of emit warning that already handles ensuring emission happens only once for the same warning.

Reply:
I think you’re thinking about deprecation warnings, which this is not (it’s just another experimental one). Anyway, I assume this is not blocking, correct?

Reply:
Ah, maybe.

Yes, I approved 🙂